src/asm/lexer.c

```diff
@@ -1520,17 +1511,167 @@ static void readString(void)
 				shiftChars(1);
 				char const *str = readMacroArg(c);
 
-				i = appendMacroArg(str, i);
+				while (*str)
+					append_yylval_tzString(*str++);
 				continue; // Do not copy an additional character
 
 			case EOF: // Can't really print that one
 				error("Illegal character escape at end of input\n");
 				c = '\\';
 				break;
+
 			default:
 				error("Illegal character escape '%s'\n", print(c));
+				shiftChars(1);
+				break;
+			}
+			break;
+
+		case '{': // Symbol interpolation
+			// We'll be exiting the string scope, so re-enable expansions
+			// (Not interpolations, since they're handled by the function itself...)
+			lexerState->disableMacroArgs = false;
+			char const *ptr = readInterpolation();
+
+			if (ptr)
+				while (*ptr)
+					append_yylval_tzString(*ptr++);
+			lexerState->disableMacroArgs = true;
+			continue; // Do not copy an additional character
+
+		// Regular characters will just get copied
+		}
+
+		append_yylval_tzString(c);
+	}
+
+finish:
+	if (i == sizeof(yylval.tzString)) {
+		i--;
+		warning(WARNING_LONG_STR, "String constant too long\n");
+	}
+	yylval.tzString[i] = '\0';
+
+	dbgPrint("Read string \"%s\"\n", yylval.tzString);
+	lexerState->disableMacroArgs = false;
+	lexerState->disableInterpolation = false;
+}
+
+static size_t appendStringLiteral(size_t i)
+{
+	dbgPrint("Reading string\n");
+	lexerState->disableMacroArgs = true;
+	lexerState->disableInterpolation = true;
+
+	bool multiline = false;
+
+	// We reach this function after reading a single quote, but we also support triple quotes
+	append_yylval_tzString('"');
+	if (peek(0) == '"') {
+		append_yylval_tzString('"');
+		shiftChars(1);
+		if (peek(0) == '"') {
+			// """ begins a multi-line string
+			append_yylval_tzString('"');
+			shiftChars(1);
+			multiline = true;
+		} else {
+			// "" is an empty string, skip the loop
+			goto finish;
+		}
+	}
+
+	for (;;) {
+		int c = peek(0);
+
+		// '\r', '\n' or EOF ends a single-line string early
+		if (c == EOF || (!multiline && (c == '\r' || c == '\n'))) {
+			error("Unterminated string\n");
+			break;
+		}
+
+		// We'll be staying in the string, so we can safely consume the char
+		shiftChars(1);
+
+		// Handle '\r' or '\n' (in multiline strings only, already handled above otherwise)
+		if (c == '\r' || c == '\n') {
+			/* Handle CRLF before nextLine() since shiftChars updates colNo */
+			if (c == '\r' && peek(0) == '\n')
+				shiftChars(1);
+			nextLine();
+			c = '\n';
+		}
+
+		switch (c) {
+		case '"':
+			if (multiline) {
+				// Only """ ends a multi-line string
+				if (peek(0) != '"' || peek(1) != '"')
```

Comment:
Is there a way to do without `peek(1)`? Perhaps:
```c
if (peek(0) == '"') {
    append_yylval_tzString('"');
    shiftChars(1);
    if (peek(0) != '"')
        break;
    append_yylval_tzString('"');
    shiftChars(1);
}
```

Reply:
I was going to take care of more points from #708 after the major PRs merged, but sure, this can be done now.